src/promptflow-azure/promptflow/azure/operations/_fileshare_storeage_helper.py

```diff
@@ -27,7 +28,38 @@
 
 class FlowFileStorageClient(FileStorageClient):
     def __init__(self, credential: str, file_share_name: str, account_url: str, azure_cred):
-        super().__init__(credential=credential, file_share_name=file_share_name, account_url=account_url)
+        if isinstance(credential, str):
+            super().__init__(credential=credential, file_share_name=file_share_name, account_url=account_url)
+        else:
+            # need add backup token intent for file share client when using AAD credential
+            self.directory_client = ShareDirectoryClient(
+                account_url=account_url,
+                credential=credential,
+                share_name=file_share_name,
+                directory_path=ARTIFACT_ORIGIN,
+                token_intent="backup",
+            )
+            self.legacy_directory_client = ShareDirectoryClient(
```

Comment:
where is this legacy_directory_client used?

Reply:
Not sure, just copy super init function here to align with sdk v2

Reply:
remove legacy_directory_client